src/rules/hostreq.ts

```diff
@@ -0,0 +1,52 @@
+import { Predicate, RuleMatchedAt, YAMLRule } from "../lint";
+import * as semver from "semver";
+import * as _ from "lodash";
+
+export class Semver implements Predicate<any> {
+  constructor(private readonly path: string,
+              private readonly required: boolean,
+  ) {}
+
+  public test(root: any): RuleMatchedAt {
+    const value = _.get(root, this.path);
+
+    if (!value && this.required) {
+      return { matched: true, paths: [] };
+    }
+
+    if (value && !semver.valid(value)) {
+      return { matched: true, paths: [this.path] };
+    }
+
+    return { matched: false };
+  }
+}
+
+export const hostreqSemverValid: YAMLRule = {
+  name: "prop-hostreq-version-semver-valid",
+  type: "error",
+  message: "replicated_version must be a valid semver specification",
+  test: new Semver("host_requirements.replicated_version", false),
+  examples: {
+    right: [{
+      yaml: `
+---
+host_requirements:
+  replicated_version: 2.9.0
```

Comment:
`replicated_version` is a semver range i believe. this is invalid

https://github.com/replicatedhq/libyaml/blob/77a767526dc8f2d90eb97d6f476f10460731abad/host_requirements.go#L4
https://github.com/replicatedhq/libyaml/blob/77a767526dc8f2d90eb97d6f476f10460731abad/validate.go#L677